src/metal/engine.ts

```diff
@@ -50,16 +55,18 @@ export class Engine implements EngineInterface {
   }
 }
 
-let globalEngine: EngineInterface = null;
-
-export function setGlobalEngine(engine: EngineInterface): boolean {
-  if (!!globalEngine) {
-    return false;
+export class OptimizedEngine extends Engine {
```

Comment:
nit-pick: can we change this to something more self-describing? `Optimized` is more of an aspirational idea then something concrete. I suspect this will only lead to a future of `ReallyOptimizedEngine` `SuperDuperOptimizedEngine` as we learn of better techniques. Instead, maybe something such as`MutationDrivenEngine` ?